TGeant3/TGeant3.cxx

```diff
@@ -2374,7 +2385,55 @@ Double_t TGeant3::TrackLength() const
    //
    // Return the length of the current track from its origin
    //
-   return fGctrak->sleng;
+   if (!gCurrentParticleStatus) {
+      return fGctrak->sleng;
+   }
+   return fGctrak->sleng + gCurrentParticleStatus->fTrackLength;
+}
+
+//______________________________________________________________________
+Int_t TGeant3::StepNumber() const
+{
+   //
+   // Return the current number of steps
+   //
+   if (!gCurrentParticleStatus) {
+      return fGctrak->nstep;
+   }
+   return fGctrak->nstep + gCurrentParticleStatus->fStepNumber;
+}
+
+//______________________________________________________________________
+Double_t TGeant3::TrackWeight() const
+{
+   //
+   // Return the current weight
+   //
+   return fGctrak->upwght;
+}
+
+//______________________________________________________________________
+void TGeant3::TrackPolarization(Double_t &polX, Double_t &polY, Double_t &polZ) const
+{
+   //
+   // Return the current polarization
+   //
+   // TODO Extract actual polarization
```

Comment:
Oops, that I should have seen. However, I haven't found a way to extract that yet. Do you know where GEANT3 stores the current polarization?

Reply:
Hello @ihrivnac 
It is currently not clear to me how to extract the polarization in case it's a Cherenkov photon. I need to understand the Fortran code for that and figure out in which common block that info is stored (it has to be somewhere).
Anyway, for now I pushed the most recent changes.